### ISSUE TITLE
HMR causing a problem with Vue-Apollo when refreshing the browser manually after HMR was triggered

### ISSUE THREAD

**Issue by IgorSzymanski** [vueneue/vueneue#22]
Link to repo: https://github.com/IgorSzymanski/vueneue-issue

## Expected behaviour
Hot reloading doesn't break when refreshing the page.

## What happens
An error 500 shows up (details below; step 5), when refreshing the browser after HMR was triggered.

## What I'm suspecting
I think it's `@vueneue\vueneue` issue with handling `vue-apollo` in SSR. I don't think it's `vue-apollo`'s issues, cause it works just fine with Nuxt.js.
Plus the issue is not present when using Vue.js's native `npm run serve` command.

## What I'm using
* `npm -v`: 6.3.0
* `node -v`: v8.9.4
* `vue -V`: 3.0.0
* OS: Windows 10, deepin OS 15.6 (Debian-based Linux dist.)

## Steps to reproduce the issue
```
git clone https://github.com/IgorSzymanski/vueneue-issue.git
npm install
npm run ssr:serve
```

1. Go to [http://127.0.0.1:8080](http://127.0.0.1:8080), to see the project in a dev mode, so you can watch it hor-reloading.
2. Edit any project file to trigger hot reloading, for instance in the file `src\main.ts` add such line at the end of the file:
    * ```ts
        console.log('foo bar')
        ```
3. The project will reload and you will see `foo bar` in the console.
4. Refresh the page in the browser (`alt + F5`).
5. An error will appear (when you edit the project, the browser will reload again, the project will seem fine unless you refresh it again manually in the browser):
    ```
    TypeError: Cannot read property 'util' of undefined
        at getMergedDefinition (webpack-internal:///./node_modules/vue-apollo/dist/vue-apollo.esm.js:341:22)
        at ApolloProvider.prefetchComponent (webpack-internal:///./node_modules/vue-apollo/dist/vue-apollo.esm.js:1322:19)
        at ApolloProvider.prefetchComponents (webpack-internal:///./node_modules/vue-apollo/dist/vue-apollo.esm.js:1348:16)
        at ApolloProvider.prefetchAll (webpack-internal:///./node_modules/vue-apollo/dist/vue-apollo.esm.js:1381:14)
        at _callee$ (webpack-internal:///./src/plugins/apollo.ts:72:49)
        at tryCatch (C:\Users\Igor\Projects\Endivy\node_modules\regenerator-runtime\runtime.js:62:40)
        at Generator.invoke [as _invoke] (C:\Users\Igor\Projects\Endivy\node_modules\regenerator-runtime\runtime.js:296:22)
        at Generator.prototype.(anonymous function) [as next] (C:\Users\Igor\Projects\Endivy\node_modules\regenerator-runtime\runtime.js:114:21)
        at step (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:12:30)
        at _next (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:27:9)
        at eval (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:34:7)
        at new Promise (<anonymous>)
        at eval (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:7:12)
        at eval (webpack-internal:///./src/plugins/apollo.ts:95:32)
        at _callee$ (webpack-internal:///./node_modules/@vueneue/ssr-core/utils/middlewares.js:121:20)
        at tryCatch (C:\Users\Igor\Projects\Endivy\node_modules\regenerator-runtime\runtime.js:62:40)
        at Generator.invoke [as _invoke] (C:\Users\Igor\Projects\Endivy\node_modules\regenerator-runtime\runtime.js:296:22)
        at Generator.prototype.(anonymous function) [as next] (C:\Users\Igor\Projects\Endivy\node_modules\regenerator-runtime\runtime.js:114:21)
        at step (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:12:30)
        at _next (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:27:9)
        at eval (webpack-internal:///./node_modules/@babel/runtime/helpers/builtin/es6/asyncToGenerator.js:34:7)
        at new Promise (<anonymous>)
    ```

**Comment by chymz** [comment]
Due to this check: https://github.com/vueneue/vueneue/blob/master/packages/%40vueneue/vue-cli-plugin-ssr/generator/templates/apollo/src/plugins/apollo.js#L5

But if I remove it, server side will crash with this error: 

```
TypeError: Cannot redefine property: $apollo
    at Function.defineProperty (<anonymous>)
    at Function.install (node_modules/vue-apollo/dist/vue-apollo.esm.js:1978:0)
```

Same problem happen with https://github.com/Akryum/vue-cli-plugin-ssr & HMR

If anyone get a solution, I'll take it :smile:

**Comment by IgorSzymanski** [comment]
Yup, I've also checked it with Akryum's plugin and couldn't get it to work.
The error caused by `Vue.use(VueApollo)` without checking it's already installed was reported lately.
https://github.com/Akryum/vue-apollo/issues/358

**Comment by LukasBombach** [comment]
This might fix it https://github.com/Akryum/vue-apollo/pull/362

**Comment by chymz** [comment]
New version of Apollo is out, I'm working to fix it !

**Comment by chymz** [comment]
Fixed in `0.3.21`
